lib/connection.js

```diff
@@ -341,14 +341,16 @@ Connection.prototype.ready = function(n){
  * @api private
  */
 
-Connection.prototype.finish = function(id){
+Connection.prototype.finish = function(id, fn){
   assertValidMessageId(id);
-  var fn = this.onerror;
   var self = this;
-  if (!this._ready) return fn(new Error('cannot finish, connection not ready'));
+  if (!this._ready) return this.onerror(new Error('cannot finish, connection not ready'));
```

Comment:
if `._ready == false`, we will never invoke the callback here. I imagine this could create pretty hard-to-debug issues down the line.

consider this code:

```js
connection.finish('abc123', function () {
  thing.doSomeWork()
})
```